### PR TITLE
[Documentation] Chapter about legacy APIs

### PR DESCRIPTION
**Pull request by AdamKasp**
| Q               | A
| --------------- | -----
| Branch?         | 1.10
| New feature?    | no
| License         | MIT

### CODE REVIEW

File: docs/book/api/legacy_api.rst [hunk 1]
```diff
@@ -0,0 +1,10 @@
+Legacy API
+==========
+
+Whe have decided that we should rebuild out APIs and unify them with API Platform.
```
Comment (GSadee) [inline docs/book/api/legacy_api.rst:4]:
```suggestion
We have decided to rebuild our APIs and unify them with API Platform.
```

File: docs/book/api/legacy_api.rst [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+Previously we had 2 separate api for shop (`SyliusShopAPi Plugin) <https://github.com/Sylius/ShopApiPlugin>`_, and for admin (`SyliusAdminApiBundle <https://github.com/Sylius/SyliusAdminApiBundle>`_).
```
Comment (GSadee) [inline docs/book/api/legacy_api.rst:5]:
```suggestion
Previously we had 2 separate APIs for shop (`SyliusShopAPi Plugin) <https://github.com/Sylius/ShopApiPlugin>`_, and for admin (`SyliusAdminApiBundle <https://github.com/Sylius/SyliusAdminApiBundle>`_).
```

File: docs/book/api/legacy_api.rst [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+Both of them are using the `FOSRestBundle <https://github.com/FriendsOfSymfony/FOSRestBundle>`_, and make operation using commands and events.
+This approach is easy to understand and implement, but when we need to customize something we need to overwrite many files (command, event, commandHandler, EventListener etc).
```
Comment (GSadee) [inline docs/book/api/legacy_api.rst:7]:
```suggestion
This approach is easy to understand and implement, but when we need to customize something we need to overwrite many files (command, event, command handler, event listener etc).
```

File: docs/book/api/legacy_api.rst [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+Second reason to create new Sylius API from scratch it fact that the API Platform is a modern framework for API and it replaces FOSRestBundle.
```
Comment (GSadee) [inline docs/book/api/legacy_api.rst:8]:
```suggestion
The second reason to create a new Sylius API from scratch is that the API Platform is a modern framework for API and replaces FOSRestBundle.
```

File: docs/book/api/legacy_api.rst [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+We will fix security issues in our legacy APIs but all new feature will be developed only in new API.
```
Comment (GSadee) [inline docs/book/api/legacy_api.rst:9]:
```suggestion
We will fix security issues in our legacy APIs but all new features will be developed only in the new API.
```

**Review by SirDomin — APPROVED** [review]

**Review by lchrusciel — APPROVED** [review]

**Comment by lchrusciel** [comment]
Thanks, Adam! :1st_place_medal:

**Pull request merged**